### PR TITLE
Merge feat/element-nav with dev

### PR DESCRIPTION
**Pull request by astra137**

### CODE REVIEW

**Comment by puregarlic** [comment]
Looks like the CI build is failing. You need to add the latest version of vue-template-compiler to your dependencies. Try running the build locally and see what happens

File: src/games/Game.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<script>
+import ElementNav from './elements/ElementNav.vue'
+import EditElement from './elements/EditElement.vue'
+
+const save = s => localStorage.setItem('elements', JSON.stringify(s))
+const load = () => JSON.parse(localStorage.getItem('elements') || false) || [
+  {
+    slug: 'player-rovs',
+    title: 'ROVs',
+    markdown: `# ROVs\n\nThe core game mechanic is driving remotely operated vehicles around a two-dimensional map.\n`
```
Comment (puregarlic) [inline src/games/Game.vue]: How come you're using vanilla strings with `\n` instead of template literals?

File: src/games/Game.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }
+]
+
+export default {
+  name: 'Game',
+  components: {
+    ElementNav,
+    EditElement
+  },
+  props: ['game', 'slug'],
```
Comment (puregarlic) [inline src/games/Game.vue]: Let's get in the habit of using the more verbose form for props. Sure, it's type checking, but there's almost no effort involved in adding it, and it's useful inline documentation. Information on the syntax can be found [here](http://devdocs.io/vue~2/api/index#props).

File: src/games/elements/EditElement.vue [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<script>
+export default {
+  name: 'EditElement',
+  props: ['markdown'],
```
Comment (puregarlic) [inline src/games/elements/EditElement.vue]: Same comment on props here.

File: src/games/elements/ElementNav.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      this.$emit('delete', e)
+      this.showAsk = false
+    }
+  }
+}
+</script>
+
+<template>
+  <el-menu :router="true">
+    <el-dialog
```
Comment (puregarlic) [inline src/games/elements/ElementNav.vue]: Turns out there's another modal-like component that's meant for this purpose: The [Message Box](http://element.eleme.io/#/en-US/component/message-box).

File: src/games/elements/NewElement.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<script>
+import slugify from 'slugify'
+
+export default {
+  name: 'NewElement',
+  props: ['game'],
```
Comment (puregarlic) [inline src/games/elements/NewElement.vue]: Expand this prop

File: src/games/elements/ElementNav.vue [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<script>
+export default {
+  name: 'ElementNav',
+  props: ['game', 'elements'],
```
Comment (puregarlic) [inline src/games/elements/ElementNav.vue]: Expand these

File: src/App.vue [hunk 7]
```diff
@@ -1,63 +1,86 @@
...
+          <el-input v-model="form.slug" auto-complete="off"></el-input>
+        </el-form-item>
+      </el-form>
+      <span slot="footer" class="dialog-footer">
+        <el-button @click="dialogFormVisible = false">Cancel</el-button>
+        <el-button type="primary" @click="dialogFormVisible = false">Confirm</el-button>
+      </span>
+    </el-dialog>
+
+    <el-menu mode="horizontal" :router="true">
```
Comment (puregarlic) [inline src/App.vue]: Let's use the prepackaged way to do layouts in Element UI, it's basically made for the look of our application: [Container](http://element.eleme.io/#/en-US/component/container)

**Review by puregarlic — CHANGES_REQUESTED** [review]
Overall functionality looks great, just needs some minor tweaks before we can fully accept it.

**Comment by astra137** [comment]
I'll expand the props, then convert the remove-element feature to message box after that. Also the container components from Element UI.

**Review by puregarlic — APPROVED** [review]
Alright, these changes look good for now. Let’s get this merged back to dev.

**Pull request merged**